client/ayon_core/hosts/houdini/api/lib.py

```diff
@@ -998,6 +998,232 @@ def add_self_publish_button(node):
     node.setParmTemplateGroup(template)
 
 
+def get_background_images(node, raw=False):
+    """"Return background images defined inside node.
+
+    Similar to `nodegraphutils.saveBackgroundImages` but this method also
+    allows to retrieve the data as JSON encodable data instead of
+    `hou.NetworkImage` instances when using `raw=True`
+    """
+
+    def _parse(image_data):
+        image = hou.NetworkImage(image_data["path"],
+                                 hou.BoundingRect(*image_data["rect"]))
+        if "relativetopath" in image_data:
+            image.setRelativeToPath(image_data["relativetopath"])
+        if "brightness" in image_data:
+            image.setBrightness(image_data["brightness"])
+        return image
+
+    data = node.userData("backgroundimages")
+    if not data:
+        return []
+
+    try:
+        images = json.loads(data)
+        if not raw:
+            images = [_parse(_data) for _data in images]
+    except json.decoder.JSONDecodeError:
+        images = []
+
+    return images
+
+
+def set_background_images(node, images):
+    """Set hou.NetworkImage background images under given hou.Node
+
+    Similar to: `nodegraphutils.loadBackgroundImages`
+
+    """
+
+    def _serialize(image):
+        """Return hou.NetworkImage as serialized dict"""
+        if isinstance(image, dict):
+            # Assume already serialized, only do some minor validations
+            assert "path" in image
```

Comment:
Note: `AssertionError` is for tests, should not be used for real error handling.